di_website/visualisation/blocks.py

```diff
@@ -0,0 +1,20 @@
+from django.utils.functional import cached_property
+from django.forms import CharField, Media
+
+from wagtail.core.blocks import FieldBlock
+from wagtail.admin.staticfiles import versioned_static
+
+from .widgets import AceEditorInput
+
+
+class ChartBlock(FieldBlock):
```

Comment:
@edwinmp 

I'm not convinced of the benefit of using `Streamfields` as a data store unless there's really no other way (with regard to UI or nested lists for instance).

They're a performance hit, as they need to be JSONed in and out of the database, and add a lot of meta-info weight around simple data structures.

In this instance I would favour an `InlinePanel` data structure if requiring multiple JSON exports per page - which I'm not sure is necessary yet, we're still only using one JSON object per chart.

I would personally keep this simple and for now have a single field attached to the page instance, or refactor to an `InlinePanel` if you want to future proof with a repeatable structure.

A couple of other things:

- I'd use a `TextField` instead of a `CharField`, as that feels more appropriate to the data type
- I'd name the field something like `data` or `json` to make it more meaningful

Reply:
Hi @edwinmp 

I remembered the Postgres-specific JSONField after adding this comment, this would be even better suited to the data type:

https://docs.djangoproject.com/en/3.0/ref/contrib/postgres/fields/#jsonfield

Reply:
Nice, I didn't know about this one ... will switch to this.